lib/puppet/provider/package/sensu_gem.rb

```diff
@@ -15,6 +15,9 @@
     if RUBY_PLATFORM =~ /cygwin|mswin|mingw|bccwin|winse|emx/
       "#{ENV['SYSTEMDRIVE']}\\opt\\sensu\\embedded\\bin\\gem.cmd"
     else
+      # grab the newest version available of the embedded ruby
+      embedded_rvm_version = Dir.glob(File.join('/opt/sensu/embedded/lib/ruby/gems/', '*')).max { |a,b| File.ctime(a) <=> File.ctime(b) }
```

Comment:
a clever solution would you mind linking to the [documentation](https://ruby-doc.org/core-2.4.0/Comparable.html)? To those that are new to ruby this would be helpful.